File: practice/algorithms/warmup/library_fine/LibraryFine.scala
object LibraryFine extends App
{
  val consoleReader = io.Source.stdin.bufferedReader()
  val actualDate = consoleReader.readLine().split(" ").map(_.toInt)
  val expectedDate = consoleReader.readLine().split(" ").map(_.toInt)

  val yearDiff = actualDate(2) - expectedDate(2)
  val monthDiff = actualDate(1) - expectedDate(1)
  val dayDiff = actualDate(0) - expectedDate(0)

  def fee(year: Int, month: Int, day: Int): Int =
  {
    if(year > 0) return 10000
    if(year == 0 && month > 0) return month * 500
    if(year == 0 && month == 0 && day > 0) return day * 15
    return 0
  }

  println(fee(yearDiff, monthDiff, dayDiff))
}

File: practice/algorithms/implementation/library_fine/LibraryFine.js
function processData(input) {
    var lines = input.split("\n");
    var actualDate = lines[0].split(" ");
    var expectedDate = lines[1].split(" ");

    var yearDiff = parseInt(actualDate[2]) - parseInt(expectedDate[2]);
    var monthDiff = parseInt(actualDate[1]) - parseInt(expectedDate[1]);
    var dayDiff = parseInt(actualDate[0]) - parseInt(expectedDate[0]);

    console.log(fee(yearDiff, monthDiff, dayDiff));
}

function fee(year, month, day)
{
    if(year > 0) return 10000;
    if(year == 0 && month > 0) return month * 500;
    if(year == 0 && month == 0 && day > 0) return day * 15;
    return 0;
}
process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});
